enterprise/jmx-monitoring/src/main/java/io/crate/beans/QueryStats.java

```diff
@@ -84,49 +96,49 @@ void inc(long duration) {
 
-    private volatile long lastUpdateTsInMillis = System.currentTimeMillis();
-
     public QueryStats(JobsLogs jobsLogs) {
         metricByStmtType = Suppliers.memoizeWithExpiration(
-            () -> {
-                long currentTs = System.currentTimeMillis();
-                Map<StatementType, Metric> metricByCommand = createMetricsMap(jobsLogs.jobsLog(), currentTs, lastUpdateTsInMillis);
-                lastUpdateTsInMillis = currentTs;
-                return metricByCommand;
-            },
+            () -> createMetricsMap(jobsLogs.metrics(), System.currentTimeMillis()),
             1,
             TimeUnit.SECONDS
         );
     }
 
-    static Map<StatementType, Metric> createMetricsMap(Iterable<JobContextLog> logEntries, long currentTs, long lastUpdateTs) {
+    static Map<StatementType, Metric> createMetricsMap(Iterable<Metrics> metrics, long currentTs) {
         Map<StatementType, Metric> metricsByStmtType = new HashMap<>();
-        long elapsedSinceLastUpdateInMs = currentTs - lastUpdateTs;
 
-        Metric total = new Metric(0, elapsedSinceLastUpdateInMs);
-        for (JobContextLog logEntry : logEntries) {
-            if (logEntry.started() < lastUpdateTs || logEntry.started() > currentTs) {
+        // the ALL metric needs the largest interval measurements are available against so it can compute the global
+        // averages (the /sec metrics are affected by this)
+        long largestMeasurementInterval = -1;
+        Metric total = new Metric(0, 0, 0);
+        for (Metrics classifiedMetrics : metrics) {
+            Histogram histogram = classifiedMetrics.histogram();
+            if (classifiedMetrics.startTime() > currentTs) {
                 continue;
             }
-            long duration = logEntry.ended() - logEntry.started();
-            total.inc(duration);
-            metricsByStmtType.compute(classificationType(logEntry), (key, oldMetric) -> {
+            long sumOfDurations = classifiedMetrics.sumOfDurations();
+            long currentMeasurementIntervalInMs = Math.max(currentTs - classifiedMetrics.startTime(), 0);
+            if (currentMeasurementIntervalInMs > largestMeasurementInterval) {
+                largestMeasurementInterval = currentMeasurementIntervalInMs;
+            }
+            total.incMetricsAndSetMeasurementInterval(sumOfDurations, histogram.getTotalCount(), largestMeasurementInterval);
+            metricsByStmtType.compute(classificationType(classifiedMetrics.classification()), (key, oldMetric) -> {
                 if (oldMetric == null) {
-                    return new Metric(duration, elapsedSinceLastUpdateInMs);
+                    return new Metric(sumOfDurations, histogram.getTotalCount(), currentMeasurementIntervalInMs);
                 }
-                oldMetric.inc(duration);
+                oldMetric.set(sumOfDurations, histogram.getTotalCount(), currentMeasurementIntervalInMs);
```

Comment:
Afaik we can have multiple `Metrics` with the same `StatementType`, so I think in order for this to yield the correct metrics we'd have to merge the histogram instead of taking the value of the one we're iterating over